docs/configuration/http_filters/fault_filter.rst

```diff
@@ -66,6 +68,20 @@ upstream_cluster:
   the config are present in the request with the same values (or based on presence if the ``value``
   field is not in the config).
 
+.. _config_http_filters_fault_injection_downstream_cluster:
+
+downstream_cluster_specific_settings:
```

Comment:
This allows us to handle cases when we'd like to get settings based on other input params (not only downstream cluster) and make decisions based only on those, having it as a config setting directly expresses the intent.

I'd vote for explicit control via configuration and avoid unnecessary lookups (not everyone wants to do downstream cluster specific conf).